tools/clang/test/hlsl_cmds_test/batch_compile.test

```diff
@@ -0,0 +1,3 @@
+// RUN: %batch -lib-link -multi-thread %S\batch_cmds2.txt
```

Comment:
`dxc_batch` is also Windows-only, so we should evaluate porting it. It also likely makes sense to pull dxc_batch tests into their own subdirectory.